arangod/Replication2/Supervision/CollectionGroupSupervision.cpp

```diff
@@ -273,8 +301,13 @@ auto pickBestServerToRemoveFromLog(
                      // then remove leaders that are not Target leaders
                      return compareTuple(left) < compareTuple(right);
                    });
+#if false
+  // TODO i think this assertion is not correct. It is possible that the leader
```

Comment:
I agree. Removing the target leader should be fine. Even if the removed server remains set in target, the supervision will handle this in `checkLeaderSetInTarget`.